addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/RequestorJobUnitTest.java

```diff
@@ -368,6 +368,116 @@ void shouldSendMessageWithSetValues() throws IOException {
         assertThat(argument.getValue().getRequestBody().toString(), is(equalTo("aaa=bbb&ccc=ddd")));
     }
 
+    @Test
+    void shouldSendHeadersWithSetValues() throws IOException {
```

Comment:
This is only testing one header.

Reply:
So should I rename it to `shouldSendHeaderWithSetValue` or should I include more headers in it. The purpose was to check if header is being sent with the request or not which is being served by sending one header.

Reply:
More headers.

Reply:
Separate unit test that sends header with same name? What should be the expected output?

Reply:
It could be all in the same but also fine if it has its own test. They should all be sent as defined.

Reply:
Use other type, e.g. List.

Reply:
List of Map?

Reply:
Might be a little odd in the YAML since there's only one for each list entry but should work. Or it could be a List of String, splitting the name/value in the job.

Reply:
Might need to change lots of stuff to do this like if I make headers to be a list of string and that string may look like `header1:value1` and later on string manipulation will be required to set this in message. getter/setter of header will also be changed/removed

Reply:
Ok I will do that but for now I need your assistance in getting things back to normal state. I messed up big time :)